app/src/main/java/com/epfl/drawyourpath/userProfile/dailygoal/DailyGoalDao.kt

```diff
@@ -101,21 +101,33 @@ interface DailyGoalDao {
     fun insertAllPoints(points: List<PointsEntity>)
 
     /**
-     * set a new distance goal to DailyGoal and User with the corresponding id and date
+     * set a new distance, time, path goal to DailyGoal and User with the corresponding id and date
      * @param userId the id of the user
      * @param date the date of the daily goal
      * @param distanceGoal the new distance goal of the user
+     * @param timeGoal the new time goal of the user
+     * @param pathsGoal the new paths goal of the user
      * @return the new daily goal
      */
     @Transaction
-    fun updateDistanceGoal(userId: String, date: Long, distanceGoal: Double): DailyGoalEntity {
-        updateDistanceGoalUser(userId, distanceGoal)
-        insertIfDailyGoalUpdateFailed(userId, date, updateDistanceGoalDailyGoal(userId, date, distanceGoal))
+    fun updateGoals(userId: String, date: Long, distanceGoal: Double? = null, timeGoal: Double? = null, pathsGoal: Int? = null): DailyGoalEntity {

```

Comment:
Same remark here, we can use `UserGoals` to avoid a large parameter list. Also, makes it easier to add a goal without changing all the function definitions.